### PR TITLE
[Merged by Bors] - feat: port RingTheory.Polynomial.Basic

### PR DESCRIPTION
**Pull request by mattrobball**
---

- [x] depends on: #3066
- [x] depends on: #2997
[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#3066
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#3066~~
* ~~leanprover-community/mathlib4#2997~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by mattrobball** [comment]
Two results remain *very* slow

File: Mathlib/RingTheory/Polynomial/Basic.lean [hunk 1]
```diff
@@ -0,0 +1,1292 @@
...
+Also, many @'s. etaExperiment caused more time outs-/
+private theorem prime_C_iff_of_fintype {R : Type u} (σ : Type v) {r : R} [CommRing R] [Fintype σ] :
+    Prime (C r : MvPolynomial σ R) ↔ Prime r := by
+  let f (d : ℕ) := (finSuccEquiv R d).symm.toMulEquiv
+  let _coe' (d : ℕ) : CoeFun ((MvPolynomial (Fin d) R)[X] ≃* MvPolynomial (Fin (d + 1)) R)
+    (fun _ => (MvPolynomial (Fin d) R)[X] → MvPolynomial (Fin (d + 1)) R) := inferInstance
+  have that (d : ℕ) : @C R (Fin (d+1)) _ r = (f d) (Polynomial.C (@C R (Fin d) _ r)) := by
+    rw [← finSuccEquiv_comp_C_eq_C]; rfl
+  rw [(renameEquiv R (Fintype.equivFin σ)).toMulEquiv.prime_iff]
+  convert_to Prime (C r) ↔ _
```
Comment (kmill) [inline Mathlib/RingTheory/Polynomial/Basic.lean:806]: It doesn't help the speed, but 
```suggestion
  convert_to Prime (C r) ↔ _ using 2
```
saves you the `congr!` in the first case. By default `convert_to` uses `using 1`.

Comment (mattrobball) [reply Mathlib/RingTheory/Polynomial/Basic.lean:806]: Done

**Pull request closed without merging**